### PR TITLE
ci: integrate SonarCloud with 60% coverage, 5% duplication, and JaCoC…

### PR DESCRIPTION
**Pull request by SaFouNaldo**
- Added SonarCloud scan to the GitHub Actions pipeline for continuous code quality analysis.
- Linked test coverage report using JaCoCo to monitor unit test coverage, setting a minimum threshold of 60%.
- Set code duplication threshold to 5% for flexibility.
- Configured project key and organization for SonarCloud integration. Secrets for SonarCloud integration can be managed [here](https://github.com/Signify-epfl/signify-app/settings/secrets/actions).

### CODE REVIEW

**Comment by ThomasMPont** [comment]
I can't open the link you provided here.
> * Configured project key and organization for SonarCloud integration. Secrets for SonarCloud integration can be managed [here](https://github.com/Signify-epfl/signify-app/settings/secrets/actions).
I assume only you have access to it. Maybe it should stay like this.

**Comment by ThomasMPont** [comment]
Is this related to a task? This should be linked to an issue on the scrum board.

**Comment by ThomasMPont** [comment]
For the following points,
> * Linked test coverage report using JaCoCo to monitor unit test coverage, setting a minimum threshold of 60%.
> * Set code duplication threshold to 5% for flexibility.
> * Configured project key and organization for SonarCloud integration. Secrets for SonarCloud integration can be managed [here](https://github.com/Signify-epfl/signify-app/settings/secrets/actions).

I can't find / don't have access to a way to validate these changes. Can you help me make sense of this with either links or screenshots?

File: sonar-project.properties
Comment (ziyadbagh) [inline sonar-project.properties:1]: looks good to me

**Review by ziyadbagh — APPROVED** [review]

File: sonar-project.properties
Comment (ThomasMPont) [reply sonar-project.properties:1]: This is way insufficient of a review. And your sentence lacks punctuation.

Unless the issues with this merge are fixed and it's changes reviewed quickly, I'll have to revert these changes.

**Comment by SaFouNaldo** [comment]
> Is this related to a task? This should be linked to an issue on the scrum board.



> For the following points,
> 
> > * Linked test coverage report using JaCoCo to monitor unit test coverage, setting a minimum threshold of 60%.
> > * Set code duplication threshold to 5% for flexibility.
> > * Configured project key and organization for SonarCloud integration. Secrets for SonarCloud integration can be managed [here](https://github.com/Signify-epfl/signify-app/settings/secrets/actions).
> 
> I can't find / don't have access to a way to validate these changes. Can you help me make sense of this with either links or screenshots?

Well, you can open the CI.yaml and see the modifications from the previous file. Basically as @ziyadbagh asked me to help him (he didn't have the rights/permissions to do so) I disabled Automatic Analysis of SonarCloud and instead I have done it directly in the CI (GitHub Actions) to avoid conflicts that could happen.
There nothing important to review here as Github runs the worfklow, (And they are more about the setup of the project than actual code) but here are the changes that were stated in my commit : 

1. Minimum test coverage needs to be at least 60% (to allow more flexibility to merge), 80% can be achieved after.
2. For the same reason (flexibility), we increased the tolerance of duplicated code to 5% we can cleap the code gradually.
3.  To do so I followed the steps of Sonar Cloud.

Hope it clearer now.

**Pull request merged**